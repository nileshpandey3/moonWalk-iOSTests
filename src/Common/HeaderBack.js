import React from "react";
import styled from "styled-components";
import Icon from "react-native-vector-icons/FontAwesome5";

const Wrapper = styled.View`
  width: 100%;
  height: 60px;
  align-items: center;
  flex-direction: row;
`;

const Title = styled.Text`
  color: white;
  font-size: 20px;
  font-weight: bold;
  width: 50%;
  text-align: center;
`;

const BackArrowWrapper = styled.TouchableOpacity`
  width: 25%;
  align-items: center;
`;

export default ({ ScreenTitle, navigateBack }) => {
  return (
    <Wrapper>
      <BackArrowWrapper onPress={navigateBack}>
        <Icon name="long-arrow-alt-left" size={25} color="white" />
      </BackArrowWrapper>
      <Title>{ScreenTitle}</Title>
    </Wrapper>
  );
};
