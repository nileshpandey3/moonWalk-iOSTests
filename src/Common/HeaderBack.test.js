import React from "react";
import HeaderBack from "./HeaderBack";
import renderer from "react-test-renderer";

describe("HeaderBack", () => {
  it("renders correctly", () => {
    const props = {
      ScreenTitle: "$screen_title$",
      navigateBack: jest.fn()
    };
    const tree = renderer.create(<HeaderBack {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
