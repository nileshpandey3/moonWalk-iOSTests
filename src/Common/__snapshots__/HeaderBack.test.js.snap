// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBack renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "opacity": 1,
        "width": "25%",
      }
    }
  >
    <Icon
      color="white"
      name="long-arrow-alt-left"
      size={25}
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
          "width": "50%",
        },
        undefined,
      ]
    }
  >
    $screen_title$
  </Text>
</View>
`;
